### ISSUE TITLE
Analytics numbers are not accurate

### ISSUE THREAD

**Issue by praminda** [wso2/product-microgateway#600]
### Description:
Uploaded statistics for API usage is not accurate. Stats are only incremented by `1`  per each stat file upload.

### Steps to reproduce:
- Configure MGW with APIM analytics
- Keep default file rotation and upload intervals
- Invoke some API resources and observe the hit count in APIM dashboards

### OS, DB, other environment details and versions:
- OS: Linux
- Client: curl
- DB: mysql
- Env (Docker/K8s): Docker

**Comment by praminda** [comment]
When writing events to `api-usage-data.dat` file we've not set the `append = true` optional property to file channel. Which replaces old events with the new ones.
However there is a bug in ballerina 0.990.4 that doesn't honour this optional property. That will be fixed by ballerina-platform/ballerina-lang#15517

**Comment by praminda** [comment]
https://github.com/wso2/product-microgateway/blob/d6053c226bf834c55aa17f834e1ccfa5274ec3e4/components/micro-gateway-core/src/main/ballerina/gateway/data_writer.bal#L63
This line should be updated with append property,
```java
io:WritableCharacterChannel charChannel = new(io:openWritableFile(fileLocation + API_USAGE_FILE, append = true), "UTF-8");
```
